controllers/heatengine_controller.go

```diff
@@ -455,9 +455,23 @@ func (r *HeatEngineReconciler) generateServiceConfigMaps(
 	if err != nil {
 		return err
 	}
+
+	// get the password of the service user from the secret
+	password, _, err := oko_secret.GetDataFromSecret(
+		ctx,
+		h,
+		instance.Spec.Secret,
+		10,
+		instance.Spec.PasswordSelectors.Service)
+	if err != nil {
+		return err
+	}
+
 	templateParameters := map[string]interface{}{
-		"KeystonePublicURL": authURL,
-		"ServiceUser":       instance.Spec.ServiceUser,
+		"KeystonePublicURL":   authURL,
+		"ServiceUser":         instance.Spec.ServiceUser,
+		"StackDomainAdmin":    heat.StackDomainAdmin,
+		"StackDomainPassword": password,
```

Comment:
StackDomainAdminPassword. Also this should be part of secret.

Reply:
Oh, actually. This isn't quite resolved. You mean we need to add these to the osp-secret secret?

Reply:
Because this is password, we should not put it into ConfigMap. Probably adding this to osp-secret is the right way to go, because we generally keep passwords for all keystone users there.